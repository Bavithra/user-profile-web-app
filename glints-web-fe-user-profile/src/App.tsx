import React from "react";
import { BrowserRouter } from "react-router-dom";

import GlobalStyle, { AppContainer, PageContainer } from "./App.styles";
import Router from "./Router";

function App() {
  return (
    <BrowserRouter>
      <GlobalStyle />
      <AppContainer>
        <PageContainer>
          <Router />
        </PageContainer>
      </AppContainer>
    </BrowserRouter>
  );
}

export default App;
