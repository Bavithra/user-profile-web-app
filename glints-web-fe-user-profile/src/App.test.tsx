import React from "react";

import App from "./App";

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import renderer from "react-test-renderer";

const mockStore = configureStore();
let store;

const initialState = {user:{
  user: {
    name: '',
    email: '',
    age: '',
    'profile-image': '',
    'work-experience': [],
  }
}}

describe("App Block", () => {
  it("renders correctly", () => {
    store = mockStore(initialState);

    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
