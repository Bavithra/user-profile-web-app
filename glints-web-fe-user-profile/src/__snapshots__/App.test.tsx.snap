// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Block renders correctly 1`] = `
<div
  className="sc-bczRLJ eiwdIu"
>
  <div
    className="sc-gsnTZi eTWJbb"
  >
    <div
      className="sc-himrzO ivdkva"
    >
      <div>
        <div
          className="sc-eCYdqJ izpdzp"
        >
          <img
            alt=""
            className="sc-jSMfEi gsWNBv"
            src="avatar.png"
          />
          <button
            className="sc-iBkjds hqHqVC"
            disabled={false}
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-xs "
              color="white"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             Upload Image
          </button>
          <input
            accept="image/*"
            className="sc-gKXOVf ixjTOr"
            data-testid="file-input"
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
      <div
        className="sc-gXmSlM iUhnBC"
      >
        <div
          className="sc-cCsOjp bWZBNe"
        >
          <div
            className="sc-jqUVSM iZwDiP"
          >
            <input
              className="sc-iqcoie dtCliV"
              maxLength={200}
              min="0"
              onChange={[Function]}
              readOnly={false}
              type="string"
              value=""
            />
            <label
              className="sc-kDDrLX eLBSyp"
            >
              Name*
            </label>
          </div>
          <div
            className="sc-jqUVSM iZwDiP"
          >
            <input
              className="sc-iqcoie dtCliV"
              maxLength={200}
              min="0"
              onBlur={[Function]}
              onChange={[Function]}
              readOnly={false}
              type="string"
              value=""
            />
            <label
              className="sc-kDDrLX eLBSyp"
            >
              Email*
            </label>
          </div>
          <div
            className="sc-jqUVSM iZwDiP"
          >
            <input
              className="sc-iqcoie dtCliV"
              maxLength={200}
              min="0"
              onChange={[Function]}
              readOnly={false}
              type="number"
            />
            <label
              className="sc-kDDrLX eLBSyp"
            >
              Age*
            </label>
          </div>
        </div>
        <div
          className="sc-papXJ dCVwsM"
        >
          Please save all the changes you have made to the profile by clicking the below button
        </div>
        <button
          className="sc-iBkjds ZgHTu"
          disabled={true}
          onClick={[Function]}
        >
          SAVE CHANGES
        </button>
        <h2
          className="sc-dIouRR iyTmtZ"
        >
          WORK EXPERIENCE
        </h2>
        <div
          className="sc-dmRaPn iGrUii"
        >
          77.9% of employers surveyed consider work experience to be a crucial data point in job applications. So be sure to fill up this section to raise your chances of securing an interview!
          <button
            className="sc-hHLeRK iNHRpL"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle-plus fa-sm "
              data-icon="circle-plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 368C269.3 368 280 357.3 280 344V280H344C357.3 280 368 269.3 368 256C368 242.7 357.3 232 344 232H280V168C280 154.7 269.3 144 256 144C242.7 144 232 154.7 232 168V232H168C154.7 232 144 242.7 144 256C144 269.3 154.7 280 168 280H232V344C232 357.3 242.7 368 256 368z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Add Work Experience
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
